python/lsst/daf/butler/registry/datasets/byDimensions/_manager.py

```diff
@@ -317,6 +317,21 @@ def register_dataset_type(self, dataset_type: DatasetType) -> bool:
             raise ValueError(
                 f"Component dataset types can not be stored in registry. Rejecting {dataset_type.name}"
             )
+
+        # If database universe and dimension group universe are different it
+        # can cause unexpected effects.
+        db_universe_version = (self._dimensions.universe.namespace, self._dimensions.universe.version)
+        dataset_type_universe_version = (
+            dataset_type.dimensions.universe.namespace,
+            dataset_type.dimensions.universe.version,
+        )
+        if dataset_type_universe_version != db_universe_version:
```

Comment:
It should also work to test `self._dimensions.universe is dataset_type.dimensions`, since `DimensionUniverse` construction works very hard to never return new instances for the same universe.  This has caused trouble in some other contexts, so maybe we don't want to add more code that depends on that behavior, but I think we've got a lot of code that does assume it.

Reply:
Hmm, identity check breaks some unit tests. I think this is because of [this code](https://github.com/lsst/daf_butler/blob/main/python/lsst/daf/butler/remote_butler/_remote_butler.py#L192-L196) in RemoteButler:

        universe = DimensionUniverse(
            config,
            # The process-global cache internal to DimensionUniverse can mask
            # problems in unit tests, since client and server live in the same
            # process for these tests.  We are doing our own caching, so we
            # don't benefit from it.  So just disable it.
            use_cache=False,
        )

Reply:
I think I found a reasonable workaround that allows identity check to work. @TallJimbo, could you re-review it?

Reply:
Looks good.

And while I don't know much about the multiple-remote-butlers scenario, I suspect that anyone who trips this tighter check by passing an equivalent but different universe instance is probably doing something less than ideal anyway.